.github/workflows/release.yml

```diff
@@ -224,17 +196,44 @@ jobs:
         with:
           signing_key_file: ${{ secrets.SIGNING_KEY_FILE }}
           signing_file_path: ${{ env.SIGNING_KEY_FILE_PATH }}
-      - name: Set github user
-        uses: ./.github/actions/set_github_user
-      - name: Pull updated files
+      - name: Change Version
         run: |
-          git pull
-      - name: Publish PayPal DataCollector
+          ./gradlew -PversionParam=${{ github.event.inputs.version }} changeReleaseVersion
+      - name: PPublish PayPal DataCollector
         uses: ./.github/actions/publish_module
         with:
           module: PayPalDataCollector
           sonatype_usr: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
           sonatype_pwd: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
           signing_key_id: ${{ secrets.SIGNING_KEY_ID }}
           signing_key_pwd: ${{ secrets.SIGNING_KEY_PASSWORD }}
-          signing_key_file: ${{ env.SIGNING_KEY_FILE_PATH }}
\ No newline at end of file
+          signing_key_file: ${{ env.SIGNING_KEY_FILE_PATH }}
+
+  releasing_finished:
+    needs: [ publish_checkout, publish_card, publish_core, publish_data_collector ]
+    name: Releases Finished
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "Release finished"
+
+  #Once all releases are done, we bump version, tag it and prepare next
+  bump_version:
+    needs: [ releasing_finished ]
+    name: Bump Version
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Set github user
+        uses: ./.github/actions/set_github_user
+      - name: Update Version
+        run: |
+          ./gradlew -PversionParam=${{ github.event.inputs.version }} changeReleaseVersion
```

Comment:
I think this should run before the releases right? We want the updated version number to be in the gradle files when released?

Reply:
@sarahkoop well here is where it gets kinda tricky:
Every `job` that publishes a module, updates the gradle file with the new version and releases. As each job runs on a diferent environment, after it completes, the changes are lost. So, once all the publishing is done, we do the "oficial" change of version in the `bump_version job` that also commits its and tags it. We decided to move the tagging and commiting to the end of the workflow, so if the publishing fails, we dont bump the version on the repo. (before it was the other way around, we commited, and then we published)

Reply:
Ok that makes sense